cdap-app-templates/cdap-dq/src/main/java/co/cask/cdap/dq/AggregationsService.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.dq;
+
+import co.cask.cdap.api.annotation.Property;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.api.service.AbstractService;
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceContext;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import co.cask.cdap.dq.functions.BasicAggregationFunction;
+import co.cask.cdap.dq.functions.CombinableAggregationFunction;
+import co.cask.cdap.dq.rowkey.AggregationsRowKey;
+import co.cask.cdap.dq.rowkey.ValuesRowKey;
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Service for querying values in the data quality histogram
+ */
+public class AggregationsService extends AbstractService {
+  public static final String SERVICE_NAME = "AggregationsService";
+  private static final Gson GSON = new Gson();
+  private static final Type TOKEN_TYPE_SET_AGGREGATION_TYPE_VALUES =
+    new TypeToken<HashSet<AggregationTypeValue>>() { }.getType();
+
+  private final String datasetName;
+
+  public AggregationsService(String datasetName) {
+    this.datasetName = datasetName;
+  }
+
+  @Override
+  protected void configure() {
+    setName(SERVICE_NAME);
+    setDescription("Service to query data quality histogram.");
+    addHandler(new ValuesLookup(datasetName));
+  }
+
+  /**
+   * Handler class for Data Quality Combinable Aggregations Service
+   */
+  @Path("/v1")
```

Comment:
Not sure about the reasoning since I didn't write this code. But I am assuming it will allow for future evolutions of the endpoints?